Catenate/Numbered_Suffix, Fopen

flowers9 opened the pull request:
Modified Catenate to be more general (now uses stdarg, and takes a
variable argument list).  Renamed Numbered_Suffix to Int_To_Str, only
takes one argument.  Fopen now makes several attempts to open a file,
with 30 second pauses in between, in order to handle possible nfs delays.

Also, Strdup and Root have been changed to use const chars for their
arguments.

Commit:
Catenate/Numbered_Suffix, Fopen

Modified Catenate to be more general (now uses stdarg, and takes a
variable argument list).  Renamed Numbered_Suffix to Int_To_Str, only
takes one argument.  Fopen now makes several attempts to open a file,
with 30 second pauses in between, in order to handle possible nfs delays.

Also, Strdup and Root have been changed to use const chars for their
arguments.

pb-cdunn commented:
- We need to make as few changes as possible, so that we can continue to merge upstream changes from thegenemeyers. We want only necessary changes.
- You dropped our new `SIZE_UNIT_LG` parameter (and didn't even mention it in your message). We need that in order to force a small test-case to be split into multiple jobs, for quick workflow testing.
- I like your `Init_DB()`.
- I _like_ GM's use of `""`.
- I don't like passing `NULL` without a good reason.
- I don't see a need for varags.
- But it doesn't matter. What matters is what GM likes.
- I assume GM has avoided `const char*` for a reason, but if `char*` is causing compiler warnings, we could try that.

You are welcome to submit these changes upstream. We really cannot accept them here.

However, you mentioned changes in daligner to accept long lines in .fasta input. We might be able to accept that. It's worth submitting.

Commit:
Revert "Catenate/Numbered_Suffix, Fopen"

This reverts commit e0283b377d2a5e861652b89c831441a24d489656.

Commit:
fasta2DB enhancement - longer lines, fastq format, compressed files

It's a bit of overkill, but I've add my open_compressed package
that handles forking off a process to read bzip2/gzip/compress files.
It's some buffering on top of read(2) calls.  fasta2DB now uses it instead
of fopen to read fasta files, and some addition logic lets fasta2DB also
read fastq files and arbitrarily long sequence lines (header lines are
still limit to MAX_NAME characters).

flowers9 commented:
Ah, I was unaware of what you guys were looking for.  Well, I reverted all those changes, and added a few things to fasta2DB.c that allow it to handle arbitrarily long sequence lines, fastq files, and compressed files.  Yours if you care for them.

pb-cdunn commented:
Thanks! I like these changes a lot, and I'm not worried about merging these. An extra file is always easy to merge, and I don't expect GM to make many changes to fasta2DB.

If you squash these and `push -f` onto your master branch, I'll merge this as-is. Otherwise, I'll do that myself. (Your name will be the "author" either way.)

Commit:
Added open_compressed to makefiles

Whoops, forgot to commit the makefile changes.  This adds open_compressed
to them so it gets linked properly.

flowers9 commented:
Ah, if you could do that, that'd be great.  I'm barely git literate as it is (and would not want to accidentally delete all the changes I made ;).

I also just pushed out the makefile changes that I forgot to commit earlier (so the new file gets linked in properly).

pb-cdunn commented:
I'm traveling Monday. This might have to wait until Tuesday.

pb-cdunn commented:
PacificBiosciences/DAZZ_DB#4
